theano/tensor/nnet/abstract_conv.py

```diff
@@ -1565,6 +1565,7 @@ def conv(self, img, kern, mode="valid", dilation=1, unshared=False):
                             if unshared is True:
                                 for row in xrange(kern.shape[2]):
                                     for col in xrange(kern.shape[3]):
+                                        val = _valfrommode('valid')
                                         out[b, n, row, col] += _convolve2d(img[b, im0, row:row + kern.shape[4],
                                                                            col:col + kern.shape[5]],
                                                                            dilated_kern[n, im0, row, col, ...],
```

Comment:
I'd retained the convolve2d because it would eliminate two more loops, and because then I would need to flip the filter manually. But I've made it element-wise now.